File: tyama_codeeval51.rb
#!/usr/bin/ruby
while (n=gets.to_i)>0
a=n.times.map{gets.split.map(&:to_i)}
r=99999
0.step(a.size-2){|i|
	(i+1).step(a.size-1){|j|
		r=[r,Math.hypot(a[i][0]-a[j][0],a[i][1]-a[j][1])].min
	}
}
$><<(r<10000?"%.4f\n":"INFINITY\n")%r
end

File: codeeval/tyama_codeeval51.c
x[99],y[99];
main(n,i,j){
	for(;scanf("%d",&n),n;){
		double R=1e8,r;
		for(i=0;i<n;i++)scanf("%d%d",x+i,y+i);
		for(i=0;i<n-1;i++)for(j=i+1;j<n;j++)if(R>(r=hypot(x[i]-x[j],y[i]-y[j])))R=r;
		printf(R<10000?"%.4f\n":"INFINITY\n",R);
	}
}